lib/mgmt_be_client.c

```diff
@@ -964,27 +923,41 @@ static void be_client_handle_notify(struct mgmt_be_client *client, void *msgbuf,
 {
 	struct mgmt_msg_notify_data *notif_msg = msgbuf;
 	struct nb_node *nb_node;
-	char notif[XPATH_MAXLEN];
 	struct lyd_node *dnode;
+	const char *data;
+	const char *notif;
 	LY_ERR err;
 
 	debug_be_client("Received notification for client %s", client->name);
 
-	err = yang_parse_notification(notif_msg->result_type,
-				      (char *)notif_msg->result, &dnode);
-	if (err)
+	notif = mgmt_msg_native_xpath_decode(notif_msg, msg_len);
+	data = mgmt_msg_native_data_decode(notif_msg, msg_len);
```

Comment:
Probably better to use `mgmt_msg_native_xpath_data_decode` here? In addition to less code, this would be the first place where this function is used, which will improve code coverage when testing.

Reply:
ok